File: sdk/r/R/dns.R
#' DNS: zones and records management for services
#' @param http HttpClient object
#' @return Resource object
#' @export
DnsResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "DnsResource"
  env
}

#' GET /api/dns - List all DNS zones
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
list <- function(self, ...) {
  self$http$get("/api/dns", list(...))
}

#' GET /api/service/{service_id}/dns - List DNS zones for service
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listForService <- function(self, ...) {
  self$http$get("/api/service/{service_id}/dns", list(...))
}

#' POST /api/service/{service_id}/dns - Create DNS zone
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
addZone <- function(self, ...) {
  self$http$post("/api/service/{service_id}/dns", list(...))
}

#' GET /api/service/{service_id}/dns/{zone_id} - Get DNS zone details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getZone <- function(self, ...) {
  self$http$get("/api/service/{service_id}/dns/{zone_id}", list(...))
}

#' DELETE /api/service/{service_id}/dns/{zone_id} - Remove DNS zone
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteZone <- function(self, ...) {
  self$http$delete("/api/service/{service_id}/dns/{zone_id}", list(...))
}

#' POST /api/service/{service_id}/dns/{zone_id}/records - Add DNS record
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
addRecord <- function(self, ...) {
  self$http$post("/api/service/{service_id}/dns/{zone_id}/records", list(...))
}

#' PUT /api/service/{service_id}/dns/{zone_id}/records/{record_id} - Edit DNS record
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
editRecord <- function(self, ...) {
  self$http$put("/api/service/{service_id}/dns/{zone_id}/records/{record_id}", list(...))
}

#' DELETE /api/service/{service_id}/dns/{zone_id}/records/{record_id} - Remove DNS record
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
deleteRecord <- function(self, ...) {
  self$http$delete("/api/service/{service_id}/dns/{zone_id}/records/{record_id}", list(...))
}

File: sdk/c/src/dns.c
#include "hitechcloud/dns.h"
#include <stdlib.h>


/* GET /api/dns - List all DNS zones */
char *hc_dns_list(hc_http_client_t *client) {
    return hc_get(client, "/api/dns", NULL);
}

/* GET /api/service/{service_id}/dns - List DNS zones for service */
char *hc_dns_list_for_service(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{service_id}/dns", NULL);
}

/* POST /api/service/{service_id}/dns - Create DNS zone */
char *hc_dns_add_zone(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{service_id}/dns", NULL);
}

/* GET /api/service/{service_id}/dns/{zone_id} - Get DNS zone details */
char *hc_dns_get_zone(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{service_id}/dns/{zone_id}", NULL);
}

/* DELETE /api/service/{service_id}/dns/{zone_id} - Remove DNS zone */
char *hc_dns_delete_zone(hc_http_client_t *client) {
    return hc_delete(client, "/api/service/{service_id}/dns/{zone_id}", NULL);
}

/* POST /api/service/{service_id}/dns/{zone_id}/records - Add DNS record */
char *hc_dns_add_record(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{service_id}/dns/{zone_id}/records", NULL);
}

/* PUT /api/service/{service_id}/dns/{zone_id}/records/{record_id} - Edit DNS record */
char *hc_dns_edit_record(hc_http_client_t *client) {
    return hc_put(client, "/api/service/{service_id}/dns/{zone_id}/records/{record_id}", NULL);
}

/* DELETE /api/service/{service_id}/dns/{zone_id}/records/{record_id} - Remove DNS record */
char *hc_dns_delete_record(hc_http_client_t *client) {
    return hc_delete(client, "/api/service/{service_id}/dns/{zone_id}/records/{record_id}", NULL);
}

